packages/led_driver/include/hardware_test_led/hardware_test_led.py

```diff
@@ -0,0 +1,125 @@
+import rospy
+import numpy as np
+
+from typing import List, Tuple, Dict
+from colorir import HSV
+
+from rgb_led import RGB_LED
+from dt_duckiebot_hardware_tests import HardwareTest, HardwareTestJsonParamType
+
+
+class HardwareTestLED(HardwareTest):
+    def __init__(
+        self,
+        driver: RGB_LED,
+        info_str: str,
+        led_ids: List[int],
+        idle_lighting: Dict[str, List],
+        fade_in_secs: int = 1,
+        duration: int = 6,
+        fade_out_secs: int = 1,
+    ) -> None:
+        # describe this group of LEDs, e.g. "front" or "back"
+        self._info_str = info_str
+        super().__init__(service_identifier=f"tests/{self._info_str}")
+
+        # attr
+        self._driver = driver
+        self._led_ids = led_ids
+        self._idle_lighting = idle_lighting  # set to this after the test
+
+        # test settings
+        self.fade_in_secs = fade_in_secs
+        self.duration = duration
+        self.fade_out_secs = fade_out_secs
+        self._color_sequence = None  # lazy init. If test is run, generate this
+
+    def test_id(self) -> str:
+        return f"LED ({self._info_str})"
+
+    def test_description_preparation(self) -> str:
+        return self.html_util_ul(
+            [
+                f"Put your Duckiebot in its normal orientation, where you can see the {self._info_str} LEDs.",
+            ]
```

Comment:
Place your Duckiebot on a flat surface in a position that allows you to see the {self._info_str} LEDs.